### PR TITLE
Omnibar Refactor: Use LegacyOmnibar

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL: https://app.asana.com/0/1174433894299346/1208319036523674

### Description
This PR continues with the refactoring, moving logic from the BTF into the Omnibar facade.

### Steps to test this PR
Smoke test of the app, chrome customtab, etc..

### CODE REVIEW

**Comment by malmstein** [comment]
* **#5073** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5073?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @malmstein and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: app/src/main/java/com/duckduckgo/app/browser/omnibar/Omnibar.kt [hunk 1]
```diff
@@ -70,28 +118,378 @@ class Omnibar(
...
+
+        customTabToolbarContainer.customTabTitle.show()
+        customTabToolbarContainer.customTabDomainOnly.hide()
+        customTabToolbarContainer.customTabDomain.show()
+        customTabToolbarContainer.customTabShieldIcon.isInvisible = showDuckPlayerIcon
+        customTabToolbarContainer.customTabDuckPlayerIcon.isVisible = showDuckPlayerIcon
+    }
+}
+
+fun EditText.replaceTextChangedListener(textWatcher: TextChangedWatcher) {
```
Comment (0nko) [inline app/src/main/java/com/duckduckgo/app/browser/omnibar/Omnibar.kt:492]: Should this be in `EditTextExtensions`?

**Review by 0nko — CHANGES_REQUESTED** [review]
When testing the changes, I noticed that when you switch the omnibar position and come back to the browser window, the find-in-page mode is incorrectly triggered. Otherwise, it seems to be working as before.

https://github.com/user-attachments/assets/4dc7a7ae-1fc4-4039-ab11-6ac96bc36e78

File: app/src/main/java/com/duckduckgo/app/browser/omnibar/Omnibar.kt (hunk 1 above)
Comment (malmstein) [reply app/src/main/java/com/duckduckgo/app/browser/omnibar/Omnibar.kt:492]: good idea, moving things around

**Review by 0nko — APPROVED** [review]
Nice work! :shipit:

**Pull request merged**